### ISSUE TITLE
Inconsistent condition between #address_offsets in elffile.py and in readelf.c

### ISSUE THREAD

**Issue by david942j** [eliben/pyelftools#138]
In method [ELFFile#address_offsets](https://github.com/eliben/pyelftools/blob/8cd87bec6f69102726a50fecbd8f5748a7cd3a75/elftools/elf/elffile.py#L131)
```python
if (start >= seg['p_vaddr'] and
   end <= seg['p_vaddr'] + seg['p_filesz']):
```

According to binutils/readelf.c#offset_from_vma:
```cpp
static long
offset_from_vma (FILE * file, bfd_vma vma, bfd_size_type size)
{
  Elf_Internal_Phdr * seg;

  if (! get_program_headers (file))
  {
      warn (_("Cannot interpret virtual addresses without program headers.\n"));
      return (long) vma;
  }

  for (seg = program_headers;
       seg < program_headers + elf_header.e_phnum;
       ++seg)
  {
      if (seg->p_type != PT_LOAD)
        continue;

      if (vma >= (seg->p_vaddr & -seg->p_align)
          && vma + size <= seg->p_vaddr + seg->p_filesz)
        return vma - seg->p_vaddr + seg->p_offset;
  }

  warn (_("Virtual address 0x%lx not located in any PT_LOAD segment.\n"),
        (unsigned long) vma);
  return (long) vma;
}
```

So seems the condition in `address_offsets` should be
```python
if (start >= seg['p_vaddr'] & -seg['p_align'] and
   end <= seg['p_vaddr'] + seg['p_filesz']):
```
Not sure what this difference affects, just figure this out.

**Comment by david942j** [comment]
In addition, `address_offsets` should add condition that only consider `LOAD` segment:
```python
for seg in self.iter_segments():
    if seg['p_type'] != 'PT_LOAD':
        continue
    if (start >= seg['p_vaddr'] and
        end <= seg['p_vaddr'] + seg['p_filesz']):
        yield start - seg['p_vaddr'] + seg['p_offset']
```

**Comment by eliben** [comment]
Thanks. It would be great if you had a test case that demonstrates this

**Comment by david942j** [comment]
I sent a pr for fixing the second comment.

For the original issue, I don't know why readelf did that (and for me that behavior seems no sense)

This issue can be closed if #159 got merged. 😄
